Source/FullScreenPokemon.ts

```diff
@@ -9196,6 +9196,38 @@ module FullScreenPokemon {
         }
 
         /**
+         * Pushes and saves the current state of a variable to a stack.
+         * 
+         * @param thing   The Thing, Area, Map, or Location saving its state of a variable.
+         * @param title   Name for the state being saved.
+         * @param value   The values of the variable to be saved.
+         */
+        addStateHistory(thing: IThing | IArea | IMap | ILocation, title: string, value: any): void {
+            if (!thing.state) {
+                thing.state = {};
+            }
+            var stateHistory: any[] = thing.state[title];
+            if (stateHistory) {
+                stateHistory.push(value);
+            } else {
+                thing.state[title] = [value];
+            }
+        }
+
+        /**
+         * Updates to the most recently saved state for a variable.
+         * 
+         * @param thing   The Thing having its state restored.
+         * @param title   The name of the state to restore.
+         */
+        popStateHistory(thing: IThing | IArea | IMap | ILocation, title: string): void {
+            var stateHistory: any[] = thing.state[title];
+            if (stateHistory && stateHistory.length > 0) {
+                thing[title] = stateHistory.pop();
+            }
```

Comment:
Can you throw an error if this is an invalid operation? There should never be a point where there isn't a state to pop from; it implies coder error.